x-pack/plugins/ml/public/application/jobs/jobs_list/components/datafeed_chart_flyout/datafeed_chart_flyout.tsx

```diff
@@ -121,6 +121,7 @@ export const DatafeedChartFlyout: FC<DatafeedChartFlyoutProps> = ({
   const [endDate, setEndDate] = useState<any>(moment(end));
   const [isLoadingChartData, setIsLoadingChartData] = useState<boolean>(false);
   const [bucketData, setBucketData] = useState<ChartDataWithNullValues>([]);
+  const [allEmptyValues, setAllEmptyValues] = useState<boolean>(false);
```

Comment:
`allEmptyValues` is always derived from `chartBucketData`, hence you don't need a dedicated state for it. Also, I find the `allEmptyValues` name a bit confusing because it does not contain all empty values, but instead is just a boolean indicator 🤔 
```suggestion
  const hasOnlyEmptyValues = useMemo(() => !chartBucketData.some((datum) => datum[1] !== null && datum[1] !== 0), [bucketData])
```

Reply:
Good call on the naming 👍 
Initially used `useMemo` but was still seeing this sometimes - dug a bit deeper due to your suggestions and looks like I missed also checking the source data. 
Now using `hasOnlyEmptyValues` for clarity and checking both bucket and source data.
Moved away from dedicated state and renamed in 44210b03c3e084648fc06068d7effd5c32e35de6